ql/src/java/org/apache/hadoop/hive/ql/optimizer/stats/annotation/StatsRulesProcFactory.java

```diff
@@ -2546,19 +2547,25 @@ private void updateColStats(HiveConf conf, Statistics stats, long leftUnmatchedR
       for (ColStatistics cs : colStats) {
         colNameStatsAvailable.add(cs.getColumnName());
         int pos = jop.getConf().getReversedExprs().get(cs.getColumnName());
-        long oldRowCount = rowCountParents.get(pos);
-        double ratio = (double) newNumRows / (double) oldRowCount;
         long oldDV = cs.getCountDistint();
+
+        boolean useCalciteForNdvReadjustment
+            = HiveConf.getBoolVar(conf, ConfVars.HIVE_STATS_JOIN_NDV_READJUSTMENT);
         long newDV = oldDV;
+        if (useCalciteForNdvReadjustment) {
+          newDV = RelMdUtil.numDistinctVals(oldDV * 1.0, newNumRows * 1.0).longValue();
```

Comment:
Can `RelMdUtil.numDistinctVals` return null? Just making sure we do not need a null check.